iac-extensions/docker/src/test/java/org/sonar/iac/docker/utils/ArgumentUtilsTest.java

```diff
@@ -64,7 +78,66 @@ void shouldNotResolveVariableOrExpandableStrings(String input) {
     assertThat(token).isNull();
   }
 
+  @ParameterizedTest
+  @ValueSource(strings = {
+    "FROM foo\nARG foo=bar\nLABEL label=$foo",
+    "FROM foo\nARG foo=bar\nLABEL label=${foo}",
+    "FROM foo\nARG foo=bar\nLABEL label=${foo:-notbar}",
+    "FROM foo\nARG foo=notbar\nARG foo=bar\nLABEL label=$foo",
+    "FROM foo\nARG foo=ar\nLABEL label=b$foo"
+  })
```

Comment:
I would also add this test for `:-` :
```
FROM foo
LABEL label=${foo:-bar}
```
Another one for `:+` :
```
FROM foo
ARG foo=notbar
LABEL label=${foo:+bar}
```
And a test for nested expression :
```
FROM foo
ARG foo2=bar
LABEL label=${foo:-$foo2}
```